src/configuration.ts

```diff
@@ -1,26 +1,52 @@
-import { BigNumber } from "ethers";
-import { parseUnits } from "ethers/lib/utils";
-
-const defaultRpcHttpUrl = process.env.RPC_HTTP_URL || process.env.RPC_URL;
+import { BigNumber, getDefaultProvider, providers } from "ethers";
+import { deepCopy, parseUnits } from "ethers/lib/utils";
 
 interface Configuration {
+  /**
+   * Whether the SDK is running in production mode or not.
+   */
   readonly isProd: boolean;
-  readonly rpcHttpUrl?: string;
-  readonly network: number;
+  /**
+   * The default provider to use for all SDK operations.
+   *
+   * @desc It can be overridden by passing a provider to the SDK when using fromChain static method of the Adapter.
+   */
+  readonly defaultProvider: providers.Provider;
+
+  /**
+   * The default max iterations for supply and borrow operations.
+   * This is the number of loops that Morpho will spend in the matching engine.
+   * The recommended value is 4 for both supply and borrow.
+   */
   readonly defaultMaxIterations: {
     readonly supply: number;
     readonly borrow: number;
   };
+
+  /**
+   * The upperbound of the transaction to add compared to the estimateGas return value.
```

Comment:
```suggestion
   * The upper bound of the transaction to add compared to the estimateGas return value.
```
I'm not sure I fully understand that sentence. Happy to have an example, I find the sentence complicated to get.

Reply:
The line below says: "if the gasLimitPercent is 110%, that means that the SDK will add 10% to the estimateGas return value."